core/src/main/scala/org/apache/spark/ExecutorAllocationManager.scala

```diff
@@ -602,6 +604,21 @@ private[spark] class ExecutorAllocationManager(
     // place the executors.
     private val stageIdToExecutorPlacementHints = new mutable.HashMap[Int, (Int, Map[String, Int])]
 
+    override def onJobStart(jobStart: SparkListenerJobStart): Unit = {
+      val jobGroupId = if (jobStart.properties != null) {
+        jobStart.properties.getProperty(SparkContext.SPARK_JOB_GROUP_ID)
+      } else {
+        ""
+      }
+      val maxConcurrentTasks = conf.getInt(s"spark.job.$jobGroupId.maxConcurrentTasks",
+        Int.MaxValue)
+
+      logInfo(s"Setting maximum concurrent tasks for group: ${jobGroupId} to $maxConcurrentTasks")
+      allocationManager.synchronized {
+        allocationManager.maxConcurrentTasks = maxConcurrentTasks
```

Comment:
yeah mark is right.  after all, that is what separates a job group property from a global property for the entire spark context.

I see why this is desirable for the most common case, of just running one job at a time, but to get this to work with multiple concurrent jobs (& job groups) you need to track a map from jobGroup -> maxConcurrency, and then sum that up (handling overflow for Int.MaxValue)